src/components/auth-provider.tsx

```diff
@@ -48,8 +48,8 @@ const AuthProvider: React.FC< AuthProviderProps > = ( { children } ) => {
 		setClient( createWpcomClient( token.accessToken, locale ) );
 		if ( token.id || token.email || token.displayName ) {
```

Comment:
This `if` check is not redundant?

Reply:
And `if (error)` seems to be the best place to show an error message to the user?

Reply:
Makes sense 👍 I've added an error modal here, inspired by https://github.com/Automattic/studio/pull/856